datumaro/util/os_util.py

```diff
@@ -49,6 +62,69 @@ def walk(path, max_depth=None):
 
         yield dirpath, dirnames, filenames
 
+def copytree(src, dst):
+    # Serves as a replacement for shutil.copytree().
+    #
+    # Shutil works very slow pre 3.8
+    # https://docs.python.org/3/library/shutil.html#platform-dependent-efficient-copy-operations
+    # https://bugs.python.org/issue33671
+
+    if sys.version_info >= (3, 8):
+        shutil.copytree(src, dst)
+        return
+
+    assert src and dst
+    src = osp.abspath(src)
+    dst = osp.abspath(dst)
+
+    if not osp.isdir(src):
+        raise FileNotFoundError("Source directory '%s' doesn't exist" % src)
+
+    if osp.isdir(dst):
+        raise FileExistsError("Destination directory '%s' already exists" % dst)
+
+    dst_basedir = osp.dirname(dst)
+    if dst_basedir:
+        os.makedirs(dst_basedir, exist_ok=True)
+
+    try:
+        if sys.platform == 'windows':
+            # Ignore
+            #   B603: subprocess_without_shell_equals_true
+            #   B607: start_process_with_partial_path
+            # In this case we control what is called and command arguments
+            # PATH overriding is considered low risk
+            subprocess.check_output(["xcopy", src, dst, # nosec
+                    "/s", "/e", "/q", "/y", "/i"],
+                stderr=subprocess.STDOUT, universal_newlines=True)
+        elif sys.platform == 'linux':
+            # As above
+            subprocess.check_output(["cp", "-r", src, dst], # nosec
+                stderr=subprocess.STDOUT, universal_newlines=True)
+        else:
+            shutil.copytree(src, dst)
+    except subprocess.CalledProcessError as e:
+        raise Exception("Failed to copy data. The command '%s' "
```

Comment:
Should probably use `IOError` for this, to mimic `copytree`.

Reply:
I'm not really sure about `OSError` (= `IOError`), because it has too much unrelated info.